### PR TITLE
radicle-git init

### PR DESCRIPTION
**Pull request by FintanH**
I am porting over all the git portions of radicle-link. Currently, a work in progress as I'm running into some strange test failures. Also some other meta setup of the repository.

### CODE REVIEW

**Comment by kim** [comment]
Hi. You are obviously free to move this code wherever you want, but could you please leave my name as attribution and copyright in place?

**Comment by FintanH** [comment]
> Hi. You are obviously free to move this code wherever you want, but could you please leave my name as attribution and copyright in place?

Sure thing! Didn't plan on removing it :)

**Comment by cloudhead** [comment]
While we're here, can we remove the part about the linking exception in the headers?

**Comment by cloudhead** [comment]
I don't know if it's included in here yet, but one very useful module is `librad::git::types` and all its descendants.

**Comment by FintanH** [comment]
> I don't know if it's included in here yet, but one very useful module is librad::git::types and all its descendants.

There's a lot of reliance on `Urn` in there when it comes to the `Remote` API. Do you have an idea to what extent you use `librad::git::types`? I'm wondering if it's just the reference API, which might be easier to massage.

**Comment by FintanH** [comment]
> >I don't know if it's included in here yet, but one very useful module is librad::git::types and all its descendants.
>
> There's a lot of reliance on Urn in there when it comes to the Remote API. Do you have an idea to what extent you use librad::git::types? I'm wondering if it's just the reference API, which might be easier to massage.

Alternatively, I could port `Urn` over from `link-identities` into `radicle-git`?

**Comment by FintanH** [comment]
I can't remember, do the github actions need to be merged to actually execute the first time? :thinking:

**Comment by keepsimple1** [comment]
@FintanH it will be helpful to add a basic `README.md` that describes the purpose and the (current planned) scope of this repo.

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  mirroring the dependency relationships of their respective "parent" crates.
+
+- `gen` and `helpers` modules are guarded behind a feature flag "test", ie.
+
+        #[cfg(any(test, feature = "test"))]
+
+- Additional helpers can be found in the `test-helpers` (preferably-pure) and
+  `it-helpers` (stateful) crates.
+
+- This crate (`tests`) does not contain any code, but depends on all other test
```
Comment (keepsimple1) [inline test/README.md:43]: Typo? This crate is named `test`, not `tests`.

Comment (FintanH) [reply test/README.md]: Fixed :)

**Comment by FintanH** [comment]
@cloudhead how's this looking for you?

**Comment by cloudhead** [comment]
Yeah, looks good. I think from here we need to basically:
* Make an inventory of the functionality we have
* Figure out what still makes sense or doesn't
* Look at the public types and figure out if we can converge towards fewer types and methods
* Look at the crates and see if they all still make sense individually

**Comment by cloudhead** [comment]
I'm also still undecided on the git-transport stuff (I don't see it here), but trying at the moment to do without and use basic git fetch/push.

**Pull request merged**